moveit_experimental/moveit_jog_arm/src/jog_cpp_interface.cpp

```diff
@@ -221,4 +221,13 @@ bool JogCppInterface::getCommandFrameTransform(Eigen::Isometry3d& transform)
   // All zeros means the transform wasn't initialized, so return false
   return !transform.matrix().isZero(0);
 }
+
+StatusCode JogCppInterface::getJoggerStatus()
```

Comment:
Why not provide a private `setJoggerStatus` so you don't have to pass the mutex around?

Reply:
I'm not sure what you mean, but I could make the status be a std::atomic<__int8> so I don't even need the mutex for this variable. How about that?

Reply:
atomic sounds good. Is there any reason why you would only make status atomic or would it make more sense for the entire shared_variables_ to be an `atomic<JogArmShared>`?

The other option that I originally suggested was to add a `std::mutex lock;` to `JogArmShared` so that you didn't have to pass the mutex around through function interfaces right next to the JogArmShared object all over the place.

A third option, if you want to get fancy would be to take care of locking and unlocking the mutex within the JogArmShared object by using setters and getters for it's internal data. You would probably make it into a class and make all of it's members private and then use the setters and getters across the code base. (eg `shared_variables_.get_ok_to_publish()` instead of `shared_variables.ok_to_publish` and `shared_variables_.set_ok_to_publish(bool ok_to_publish`)